node/type.go

```diff
@@ -0,0 +1,32 @@
+package node
+
+// node.Type defines a type for Nodes to be differentiated.
+type Type uint8
+
+const (
+	// node.Full is a full-featured Celestia Node.
+	Full Type = iota + 1
```

Comment:
```suggestion
	Full Type = iota
```

Reply:
The reason I added 1 is that the zero value of `Type` would be a valid value, but we should avoid that, as having valid zero value enum is bug-prone. Instead, I could have defined `Unknown` Type/Mode for zero value, but in practice, those unknown values are just sitting in the code and no one directly uses/imports them, so I decided to omit it.

Reply:
I'll add a comment describing that

Reply:
sounds good thank u